snoop/data/management/commands/filestats.py

```diff
@@ -0,0 +1,181 @@
+
+    Args:
+        collections_list: A list of collections that will be analyzed.
+        print_supported: When False only analyzes unsupported filetypes.
+    """
+    res = {}
+    for col in collections_list:
+        collection = collections.ALL[col]
+        with collection.set_current():
+            queryset_mime = models.Blob.objects.all().values('mime_type', 'magic')\
+                .annotate(total=Count('mime_type')).annotate(size=Sum('size'))\
+                .order_by('-size')
+            if not print_supported:
+                queryset_mime = queryset_mime.exclude(mime_type__in=SUPPORTED_MIME_TYPES)
+            for mtype in queryset_mime:
+                if mtype['mime_type'] not in res:
+                    res[mtype['mime_type']] = {'size': truncate_size(mtype['size']),
+                                               'magic': get_description(col, mtype['mime_type'])}
+                else:
+                    res[mtype['mime_type']]['size'] += truncate_size(mtype['size'])
+    sorted_res = sorted(res.items(), key=lambda x: x[1]['size'], reverse=True)
+    return dict(sorted_res)
+
+
+def get_top_extensions(collections_list, print_supported=True):
+    """Return a dictionary of file extensions that occupy most space in collections.
+
+    Args:
+        collections_list: A list of collections that will be analyzed.
+        print_supported: When False only analyzes unsupported filetypes.
+    """
+    ext_dict = {}
+    for col in collections_list:
+        query = r"""select substring(encode(f.name_bytes::bytea, 'escape')::text
+                    from '(\..{1,20})$') as ext,
+                    sum(f.size) as size,
+                    b.mime_type as mime
+                    from data_file f
+                    join data_blob b on f.blob_id = b.sha3_256
+                    group by ext, mime
+                    order by size desc limit 100;"""
+        with connections[collections.ALL[col].db_alias].cursor() as cursor:
+            cursor.execute(query)
+            results = cursor.fetchall()
+
+        for ext, size, mime in results:
+            if not print_supported:
+                if mime in SUPPORTED_MIME_TYPES:
+                    continue
+            if ext not in ext_dict:
+                ext_dict[ext] = {'size': truncate_size(int(size)), 'mtype': set([mime])}
+            else:
+                ext_dict[ext]['size'] += truncate_size(int(size))
+                ext_dict[ext]['mtype'].add(mime)
+    sorted_ext_dict = sorted(ext_dict.items(), key=lambda x: x[1]['size'], reverse=True)
+    return dict(sorted_ext_dict)
+
+
+def get_description(col, mime_type, *extension):
+    """Return the magic description for a given mime-type.
+
+    Args:
+        col: Collection on which the query is executed.
+        mime_type: Mime-Type for which the descriptions is returned.
+        *extension: Optional file-extension as string to limit the search to this
```

Comment:
1. I don't see anywhere where this `*extension` argument is used -- you didn't add the mime descriptions for the second extension table, and that's OK. Since the argument is unused, I'd delete the argument and the `if extension` and simplify this function.
2. `*extension` doesn't mean optional argument, it means variadic argument list. You generally don't want to use variadic functions unless you've got some good reason to (like composition when being called by some library with arguments you don't care about, or when it's natural inside the caller code) -- here, there's no reason to: you seem to want to get an optional list of extensions (only to ignore most of them and look at the first one in the list) -- so you should use an argument with default value  `extensions=[]`. But then, if you somehow modify the list, you run into this: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments - so either do the work-around there (leave the default argument value as `=None` and then change it), or be sure not to mutate the argument. Anyway, you probably don't need this argument anyway, but I wanted you to know the problems here